src/impl_/pyclass/static_type_object.rs

```diff
@@ -24,31 +28,61 @@ pub struct LazyStaticType {
     tp_dict_filled: GILOnceCell<PyResult<()>>,
```

Comment:
The detection of re-entrant initialization seem surprisingly heavy weight. What do you think about turning this around in the future, meaning to use e.g. a thread-local list of type object addresses which the current thread is currently initializing instead of a list of threads for each type object?

Reply:
I think it's not something I'll rush to do because this code only affects program start-up time. It's a good idea and will be necessary if we ever try to replace these static things with type objects which live as data on the module. Created #2950